src/org/rascalmpl/library/Set.rsc

```diff
@@ -513,3 +513,29 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
 
 @synopsis{Compute the Jaccard similarity between two sets.}
 real jaccard(set[value] x, set[value] y) = (1. * size(x & y)) / size(x + y);
+
+
+@synopsis{Calculate the intersection of a set of sets.}
+public set[&T] intersection(set[set[&T]] sets)  = (getFirstFrom(sets) | it & elem | elem <- sets);
+
+
+@synopsis{Checks if all sets in the set are pairwise disjoined.}
+@examples{
+```rascal-shell
+import Set;
+isDisjoined({{1,2}, {3,4}, {5,6}});
+isDisjoined({{1,2}, {1,4}, {5,6}});
+isDisjoined({{1,2}, {1,4}, {1,6}});
+```
+}
+public bool isDisjoined(set[set[&T]] sets) {
+  list[set[&T]] setsAsList = toList(sets);
+
+  for (elem1 <- [0..size(setsAsList)-1]) {
+    for (elem2 <- [elem1+1..size(setsAsList)]) {
+      if (setsAsList[elem1] & setsAsList[elem2] != {}) return false;
```

Comment:
pls add a newline and brackets?

Reply:
You could avoid the code for generating all combinations by using set matching. It will produce the exact same combinations, and we still use your smart trick of reusing the fast intersection operator:
 
```
bool isDisjoint({set[&T] a, set[&T] b, *_}) = false when a & b != {}; // will backtrack to other pairs of a and b while the condition fails
default bool isDisjoint({set[&T] a, set[&T] b, *_}) = true;
```